packages/esm-patient-list-management-app/src/lists-dashboard/lists-dashboard.component.tsx

```diff
@@ -42,6 +42,16 @@ const ListsDashboard: React.FC = () => {
   const patientListFilter = usePatientListFilterForCurrentTab(selectedTab);
   const { patientLists, isLoading, error, mutate } = useAllPatientLists(patientListFilter);
   const { search } = useLocation();
+  const newCohortUrl = window.getOpenmrsSpaBase() + 'home/patient-lists?new_cohort=true';
+  const handleShowNewListOverlay = () => {
+    // URL navigation is in place to know either to open the create list overlay or not
+    // The url /patient-list?new_cohort=true is being used in the "Add patient to list" widget
+    // in the patient chart. The button in the above mentioned widget "Create new list", navigates
+    // to /patient-list?new_cohort=true to open the overlay directly.
+    navigate({
+      to: newCohortUrl,
+    });
```

Comment:
This is very fragile. This is making in-page logic dependent on the URL remaining `home/patient-lists`. It will be very annoying if someone changes that URL and then can't figure out why adding a cohort isn't working.

I suggest that instead, you add a state `showCreatePatientList`, initialize that state using the query param, have the button modify that state, and use that state instead of `isCreatingPatientList` to determine whether to show `<CreateEditPatientList>`.

Does that make sense to you?

Reply:
We had a call and I explained it.